dpnp/dpnp_iface_mathematical.py

```diff
@@ -3255,14 +3264,31 @@ def interp(x, xp, fp, left=None, right=None, period=None):
 
 """
 
-maximum = DPNPBinaryFunc(
+_maximum_impl = DPNPBinaryFunc(
```

Comment:
Would it be better to create `DPNPBinaryFuncOutKw` class inherited from `DPNPBinaryFunc` one in `dpnp/dpnp_algo/dpnp_elementwise_common.py` where `__call__` method will be wrapped by similar way.
It'd help to reduce the code duplication.